modules/stub/src/stub.js

```diff
@@ -1,211 +1,245 @@
-(function() {
+((win, doc) => {
 
   const makeStub = () => {
 
-    const TCF_LOCATOR_NAME = '__tcfapiLocator';
-    const queue = [];
-    let win = window;
-    let cmpFrame;
+    const TCF_API_NAME = '__tcfapi';
+    const EVENT_NAME = 'message';
+    const TCF_POST_MSG_CALL = TCF_API_NAME + 'Call';
+    const TCF_POST_MSG_RETURN = TCF_API_NAME + 'Return';
+    const TCF_LOCATOR_NAME = TCF_API_NAME + 'Locator';
+    const CMP_STATUS = 'stub';
+    const TARGET_ORIGIN = '*';
+    const PING_COMMAND = 'ping';
+    const GDPR_APPLIES_COMMAND = 'setGdprApplies';
 
-    function addFrame() {
+    if (!(win.frames[TCF_LOCATOR_NAME])) {
 
-      const doc = win.document;
-      const otherCMP = !!(win.frames[TCF_LOCATOR_NAME]);
+      if (doc.body) {
 
-      if (!otherCMP) {
+        const iframe = doc.createElement('iframe');
 
-        if (doc.body) {
+        /**
+         * Create the locator frame in this frame since we've established that
+         * none exists in this frame.  This is to prevent instances of this
+         * script from overriding each other.  If this one queues calls and
+         * gets overridden then the queue is lost and calls are dropped.
+         */
+        iframe.style.cssText = 'display:none';
+        iframe.name = TCF_LOCATOR_NAME;
+        doc.body.appendChild(iframe);
 
-          const iframe = doc.createElement('iframe');
+        if (win === win.top || !(win.parent.frames[TCF_LOCATOR_NAME])) {
```

Comment:
With this change, are you now only checking 1 frame up (just the parent)?  If you're looking for an existing CMP, wouldn't you need to walk all the way to the top frame?  That said,  is it necessary to check for CMPs in other frames at all?  It seems like a strange scenario to have 2 CMP instances loaded on the same page.  I would think simply checking the current frame would be sufficient, but maybe I'm missing some use case.  

Is the purpose of checking for CMPs in other frames to have this stub also act as a proxy to any other CMP that might be running on the page?  If so, it seems like a cool idea, but maybe it makes more sense separated out into a separate "proxy" module. 

In any case, I believe this resolves the issue I reported.

Reply:
I walked away from this and then came back to it, I think I left it mid thought.  I refactored yet again.

> is it necessary to check for CMPs in other frames at all? 

Yes I believe so, because if ad code is loaded into an iframe and they include this stub then it will proxy calls to the top CMP.

> Is the purpose of checking for CMPs in other frames to have this stub also act as a proxy to any other CMP that might be running on the page?

Yeah, maybe that's a better idea...